File: backend/backend/src/main/java/Shapes/Ellipse.java
package Shapes;

import HelpingClasses.Sizes.*;

public class Ellipse extends ClosedShape
{
    public Ellipse(int id)
    {
        super(id);
    }

    public void SetSize(Size size)
    {
        this.SetTriangleSize((EllipseSize) size);
    }

    private void SetTriangleSize(EllipseSize size)
    {
        this.MySize = size;
    }
}

File: frontend/paint_frontend/src/app/classes/Ellipse.ts
import { ClosedShape } from "./ClosedShape";
import { LWSize } from "./LWSize";
import { Position } from "./Position";

export class Ellipse extends ClosedShape{
    constructor(id:number, size:LWSize, strokeWidth?:number, 
        enabled?:boolean, strokeColor?:string, 
        fillColor?:string, shapePosition?: Position){

            super(id, size, strokeWidth,
                enabled, strokeColor, fillColor, shapePosition);

    }

    override setSize(size:LWSize){
        this.size = size;
    }
}